internal/cli/atlas/deployments/search/indexes/delete.go

```diff
@@ -0,0 +1,154 @@
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/atlas/deployments/options"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/atlas/search"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/log"
+	"github.com/mongodb/mongodb-atlas-cli/internal/mongodbclient"
+	"github.com/mongodb/mongodb-atlas-cli/internal/podman"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+const (
+	deletedMessage      = "Index '%s' deleted"
+	deleteMessageFailed = "Index not deleted"
+)
+
+type DeleteOpts struct {
+	cli.GlobalOpts
+	*cli.DeleteOpts
+	options.DeploymentOpts
+	search.IndexOpts
+	mongodbClient mongodbclient.MongoDBClient
+	store         store.SearchIndexDeleter
+}
+
+func (opts *DeleteOpts) Run(ctx context.Context) error {
+	if err := opts.validateAndPrompt(ctx); err != nil {
+		return err
+	}
+
+	if err := opts.Prompt(); err != nil {
+		return err
+	}
+
+	err := opts.RunLocal(ctx)
+	if err != nil && (errors.Is(err, options.ErrDeploymentNotFound)) {
+		return opts.RunAtlas()
+	}
+
+	return err
+}
+
+func (opts *DeleteOpts) RunAtlas() error {
+	return opts.Delete(opts.store.DeleteSearchIndex, opts.ConfigProjectID(), opts.DeploymentName)
+}
+
+func (opts *DeleteOpts) RunLocal(ctx context.Context) error {
+	connectionString, err := opts.ConnectionString(ctx)
+	if err != nil {
+		return err
+	}
+
+	if err = opts.mongodbClient.Connect(connectionString, connectWaitSeconds); err != nil {
+		return err
+	}
+	defer opts.mongodbClient.Disconnect()
+
+	return opts.Delete(opts.mongodbClient.DeleteSearchIndex, opts.Entry)
+}
+
+func (opts *DeleteOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *DeleteOpts) initMongoDBClient(ctx context.Context) func() error {
+	return func() error {
+		opts.mongodbClient = mongodbclient.NewClientWithContext(ctx)
+		return nil
+	}
+}
+
+func (opts *DeleteOpts) validateAndPrompt(ctx context.Context) error {
+	if opts.Entry == "" {
+		if err := promptRequiredName("Search Index ID", &opts.Entry); err != nil {
+			return err
+		}
+	}
+
+	if opts.DeploymentName == "" {
+		if err := opts.DeploymentOpts.Select(ctx); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func DeleteBuilder() *cobra.Command {
+	opts := &DeleteOpts{
+		DeleteOpts: cli.NewDeleteOpts(deletedMessage, deleteMessageFailed),
+	}
+	cmd := &cobra.Command{
+		Use:     "delete <indexId>",
+		Aliases: []string{"rm"},
+		Short:   "Delete the specified search index from the specified cluster.",
+		Long:    fmt.Sprintf(usage.RequiredRole, "Project Data Access Admin"),
```

Comment:
is this true? this says you need to be logged in I don't think that's the case

Reply:
it is true for the atlas cluster 🤔 I will update the text to make it clear

Reply:
yes, but for running the command overall it shouldn't be required I think, for atlas it would just say the same "if you want to get atlas indexes, login" right?

Reply:
yes, you are right. I removed it